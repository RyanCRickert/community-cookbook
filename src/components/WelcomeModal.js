import React from "react";
import Modal from "react-modal";

export default class PostForm extends React.Component {
	constructor(props) {
		super(props);

	}

	render() {
		return (
			<Modal
				ariaHideApp={false}
				onRequestClose={this.props.closeWelcomeModal}
				isOpen={!!this.props.firstVisit}
				closeTimeoutMS={200}
				className="modal-intro"
			>
				<div>
					<h1>Hello and Welcome</h1>
					<div className="modal-intro__body">
						Just a quick word of advice for a first time visitor.  If you would like to create an account, please feel free to do so.  Initial accounts will only 
						have the privileges of editing their own personal shopping list.  To request the ability to add or edit recipes, please contact me.  I have allowed anyone to view 
						the process if they wish.
					</div>
					<button className="button" onClick={this.props.closeWelcomeModal}>Close</button>
				</div>
      </Modal>
		)
	}
}