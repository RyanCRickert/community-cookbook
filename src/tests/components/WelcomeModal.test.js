import React from "react";
import { shallow } from "enzyme";
import WelcomeModal from "../../components/WelcomeModal";

test("should render WelcomeModal correctly", () => {
  const wrapper = shallow(<WelcomeModal/>)
  expect(wrapper).toMatchSnapshot();
});