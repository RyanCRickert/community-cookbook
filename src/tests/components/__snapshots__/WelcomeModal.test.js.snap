// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WelcomeModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal-intro"
  closeTimeoutMS={200}
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div>
    <h1>
      Hello and Welcome
    </h1>
    <div
      className="modal-intro__body"
    >
      Just a quick word of advice for a first time visitor.  If you would like to create an account, please feel free to do so.  Initial accounts will only have the privileges of editing their own personal shopping list.  To request the ability to add or edit recipes, please contact me.  I have allowed anyone to view the process if they wish.
    </div>
    <button
      className="button"
    >
      Close
    </button>
  </div>
</Modal>
`;
